test/Unit/models/Import/Service/SendCalculatedResultServiceTest.php

```diff
@@ -0,0 +1,199 @@
+
+        $outcomeVariables = $this->createVariables(false);
+
+        $return = $this->doDeliveryExecution($outcomeVariables, $qtiTestItems);
+
+        $this->assertIsArray($return);
+        $this->assertArrayHasKey('gradingStatus', $return);
+        $this->assertSame(ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED, $return['gradingStatus']);
+    }
+
+    public function testDeclarationIsScoredVariableIsGraded()
+    {
+        $qtiTestItems = $this->createDeclarations(true);
+
+        $outcomeVariables = $this->createVariables(true);
+
+        $return = $this->doDeliveryExecution($outcomeVariables, $qtiTestItems);
+
+        $this->assertIsArray($return);
+        $this->assertArrayHasKey('gradingStatus', $return);
+        $this->assertSame(ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED, $return['gradingStatus']);
+    }
+
+    private function createVariables(bool $isExternallyGraded): array
+    {
+        $data = [
+            'type' => taoResultServer_models_classes_OutcomeVariable::TYPE,
+            'normalMaximum' => null,
+            'normalMinimum' => null,
+            'value' => 'MA==',
+            'identifier' => 'OUTCOME_1',
+            'cardinality' => 'single',
+            'baseType' => 'float',
+            'epoch' => '0.80757100 1681904685',
+            'externallyGraded' => $isExternallyGraded,
+        ];
+        $variable = taoResultServer_models_classes_OutcomeVariable::fromData($data);
+
+        $container = new stdClass;
+        $container->variable = $variable;
+
+        return [
+            [
+                $container
+            ]
+        ];
+    }
+
+    private function createDeclarations($isExternallyScored): array
+    {
+        return [
+            'testPart-1' => [
+                'assessmentSection-1' => [
+                    'item-1' => [
+                        'outcomes' => [
+                            [
+                                'identifier' => 'OUTCOME_1',
+                                'attributes' =>
+                                    [
+                                        'identifier' => 'OUTCOME_1',
+                                        'externalScored' => 'externalMachine',
+                                    ],
+                            ]
+                        ],
+                        'isExternallyScored' => $isExternallyScored
+                    ]
+                ],
+            ]
+        ];
+    }
+
+    private function doDeliveryExecution(array $outcomeVariables, array $qtiTestItems): array
+    {
+        $this->rdsResultStorageMock
+            ->expects($this->any())
+            ->method('getDeliveryVariables')
+            ->willReturn($outcomeVariables);
+
+        $this->qtiRunnerInitDataBuilderMock
+            ->expects($this->any())
+            ->method('getQtiTestItems')
+            ->willReturn($qtiTestItems);
+
+        $scrs = new SendCalculatedResultService(
```

Comment:
You do not need setters to set up mocks whenever you want, cause they are stored in properties of your test and only then passed to SendCalculatedResultService constructor,  you can access them.
My point is to avoid instantiating/calling test subject and hiding expectations in private method so that make code more clear.

Reply:
Moved it and it works :) Thanks it looks better this way.